packages/vscode-dependencies-validation/src/editorChanges.ts

```diff
@@ -17,12 +18,13 @@ export function subscribeToPackageJsonChanges(
 ): void {
   const { window, subscriptions, workspace, diagnosticCollection } =
     vscodeConfig;
-
-  if (window.activeTextEditor) {
-    void refreshDiagnostics(
-      window.activeTextEditor.document.uri,
-      diagnosticCollection
-    );
+  if (
```

Comment:
Would it be better to get uri before the if and then just use it twice in the method ?

const uri  = window.activeTextEditor?.document.uri

Reply:
I tried that earlier, but than the type of `uri` would be `Uri | undefined`

![image](https://user-images.githubusercontent.com/4233843/148257706-7af24702-3b7f-4073-a1d3-17d3eb336462.png)

So I would need to adjust the code in ways that make less sense, e.g:
- nested `if` inside `if`
- additional condition to check `uri` is undefined

So lacking a "perfect" way I just choose to stick with the simplest one
Can you suggest an alternative?

Reply:
just change the if to ---> if (uri && shouldBeChecked(uri.path))